modules/caddyhttp/http2listener.go

```diff
@@ -0,0 +1,96 @@
+package caddyhttp
+
+import (
+	"context"
+	"crypto/tls"
+	weakrand "math/rand"
+	"net"
+	"net/http"
+	"sync/atomic"
+	"time"
+
+	"golang.org/x/net/http2"
+)
+
+type http2Listener struct {
```

Comment:
A comment explaining the usefulness of this type instead of just the http2 package types would be useful!